backend/src/test/java/de/neuefische/koheis/backend/gamegcards/GameCardsGridIntegrationTest.java

```diff
@@ -21,22 +24,38 @@ class GameCardsGridIntegrationTest {
     @Autowired
     GameCardsRepository gameCardsRepository;
 
+    @Test
+    void whenListEmpty_thenReturnEmptyList() throws Exception {
+        //WHEN
+        mockMvc.perform(
+                        MockMvcRequestBuilders.get("/api/game_cards/all")
+                )
+
+                //THEN
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(content().json("""
+                        []
+                        """));
+
+    }
+
+
     @Test
     void expectTwoDimensionalArrayOfTwentyEightPlayingCards() throws Exception {
         //GIVEN
         String size = "small";
-        GameCard gameCard1 = new GameCard("♥1", "playing-cards");
-        GameCard gameCard2 = new GameCard("♥2", "playing-cards");
-        GameCard gameCard3 = new GameCard("♥3", "playing-cards");
-        GameCard gameCard4 = new GameCard("♥4", "playing-cards");
-        GameCard gameCard5 = new GameCard("♥5", "playing-cards");
-        GameCard gameCard6 = new GameCard("♥6", "playing-cards");
-        GameCard gameCard7 = new GameCard("♥7", "playing-cards");
-        GameCard gameCard8 = new GameCard("♥8", "playing-cards");
-        GameCard gameCard9 = new GameCard("♥9", "playing-cards");
-        GameCard gameCard10 = new GameCard("♥10", "playing-cards");
-        GameCard gameCard11 = new GameCard("♥11", "playing-cards");
-        GameCard gameCard12 = new GameCard("♥12", "playing-cards");
+        GameCard gameCard1 = new GameCard("01", "♥1", "playing-cards");
+        GameCard gameCard2 = new GameCard("012", "♥2", "playing-cards");
+        GameCard gameCard3 = new GameCard("03", "♥3", "playing-cards");
+        GameCard gameCard4 = new GameCard("04", "♥4", "playing-cards");
+        GameCard gameCard5 = new GameCard("05", "♥5", "playing-cards");
+        GameCard gameCard6 = new GameCard("06", "♥6", "playing-cards");
+        GameCard gameCard7 = new GameCard("07", "♥7", "playing-cards");
+        GameCard gameCard8 = new GameCard("08", "♥8", "playing-cards");
+        GameCard gameCard9 = new GameCard("09", "♥9", "playing-cards");
+        GameCard gameCard10 = new GameCard("10", "♥10", "playing-cards");
+        GameCard gameCard11 = new GameCard("11", "♥11", "playing-cards");
+        GameCard gameCard12 = new GameCard("12", "♥12", "playing-cards");
 
```

Comment:
In integration tests we want to look at and treat our application from the outside. In this case, adding new cards would take place via the post endpoint. That's possible with your application, isn't it? Like this:
´´´java
 mockMvc.perform(
                        MockMvcRequestBuilders.post("/api/game_cards")
                                .contentType(MediaType.APPLICATION_JSON)
                                .content("""
                                        {
                                        "title": "testTitle",
                                        "cardSetName": "testSet3"
                                        }
                                        """)
                )´´´ And this post for every gameCard here. But as you like. Unfortunately this is a lot of work here ;D

Reply:
Your are absolutely right.  My test looks unreasonable, because I wrote unfortunately the second get method with parameters at first 🙃 The post method is the newest to be tested in this user story... Anyway, thank you for your comment 🙏 I will rewrite the test in the next user story.